frontends/p4/reassociation.cpp

```diff
@@ -25,7 +25,7 @@ const IR::Node *Reassociation::reassociate(IR::Operation_Binary *root) {
     if (!leftBin) return root;
     if (leftBin->getStringOp() != root->getStringOp()) return root;
     if (!leftBin->right->is<IR::Constant>()) return root;
-    auto c = dynamic_cast<IR::Operation_Binary *>(root->clone());
+    auto c = root->clone()->to<IR::Operation_Binary>();
```

Comment:
Might as well do the check to prevent null-deref. And I think doing the check before clone is minorly better.

```suggestion
    auto c = root->checkedTo<IR::Operation_Binary>()->clone();